File: app/src/main/java/com/yvrun/officeprocess/util/adapter/ATextTWatcher.java
package com.yvrun.officeprocess.util.adapter;

import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;

public abstract class ATextTWatcher implements TextWatcher {

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

        Log.d("--TAG--", "beforeTextChanged: " + s);
    }

    @Override
    public void afterTextChanged(Editable s) {

        Log.d("--TAG--", "beforeTextChanged: " + s);
    }
    
}

File: kotlinappother/src/main/java/com/chengzhen/kotlinappother/ATextTWatcher.kt
package com.chengzhen.kotlinappother

import android.text.Editable
import android.text.TextWatcher
import android.util.Log

abstract class ATextTWatcher : TextWatcher {

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

        Log.d("--TAG--", "beforeTextChanged: $s")
    }

    override fun afterTextChanged(s: Editable) {

        Log.d("--TAG--", "beforeTextChanged: $s")
    }

}
